chapter8.adoc

```diff
@@ -1,37 +1,124 @@
+. Triggers must match and fire on a hart only when a supervisor domain with this
+  configuration is active on a hart (and not in M-mode).
+. The RDSM must be able to context switch all triggers even when `d-mode` = 1.
+  This functional change allows the RDSM to remain in control of external debug
+  for supervisor domains when it is not under external debug itself.
+
+The following figure is non-normative and is intended to illustrate the use of
+the `msdcfg.sdedbgalw` control - the normative specification will be in the
+specification for external debug security cite:[ExtDbgSec].
 
 [caption="Figure {counter:image}: ", reftext="Figure {image}"]
-[title= "External Debug for Supervisor Domain", id=Smsdedbg_img]
+[title= "External Debug opt-in for Supervisor Domain", id=Smsdedbg_img]
 image::images/Smsdedbg.png[]
 
-This section will be moved into the non-ISA specification for external debug
-security. It is described here as informational.
-
-The `medbgen` is an enable control for external debug for the M-mode driven by
-the debug module and is expected to be established by the RoT (following RISC-V
-Security Model recommendation SR_GEN_007 and SR_GEN_012). When privilege is `M`,
-the `medbgen` gates the `haltreq` from the debug module and if is 0 prevents
-the hart from entering external debug mode. When privilege is less than `M`, the
-OR of the `MSDCFG.sdedbgalw` and `medbgen` gates the `haltreq` from the debug
-module and the hart will enter external debug mode if either is 1.
-
-The configuration for `MSDCFG.sdedbgalw` may be obtained from the manifest/
-configuration of the supervisoer domain and should be managed by the M-mode root
-security manager using secure memory.
-
-When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
-`MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but is
-restricted by `M-mode` software to prevent privileged CSR accesses and memory
-accesses by instructions executed in external debug mode cannot use `M` privilege.
-When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, the M-mode root domain
-security manager must not configure triggers with `action=1`. Triggers for the
-supervisor domain are expected to be controlled by the root domain security
-manager to prevent any leakage of information.
+=== `Smsdetrc`: External Trace allowed control for Supervisor Domain
+
+When M-mode external trace is enabled, all supervisor domains activity may also
+be traced by an external trace tool irrespective of the configuration held in
+`msdcfg.sdetralw`.
+
+When M-mode external trace is disabled, whether execution at privilege modes
+less than `M-mode` may be traced by an external trace tool depends on the
+configuration held in `msdcfg.sdetrcalw`, as described below:
+
+When `msdcfg.sdetrcalw` = 0, external trace is disallowed for the SD. All trace
+output from the trace encoder module are disabled while the SD is active on a
+per-hart basis.
+
```

Comment:
"while SD is.." can be removed. Can just state "When `msdcfg.sdetrcalw` = 0, external trace of the SD is disallowed".

Reply:
ack